### ISSUE TITLE
Log levels do not apply to the static Logger

### ISSUE THREAD

**Issue by phils-hub** [nestjs/nest#5578]
## Current behavior
Setting the log level for the standard Logger does not apply that level to the static instance. Consequently, it is not possible to define a global log level that applies to all calls to the standard logger.

## Input Code
```ts
// main.ts
const app = await NestFactory.createApplicationContext(
    AppModule,
    { 
      logger: (_ => { return ["error", "warn"] })
    }
)

// any other class
import { Injectable, Logger } from '@nestjs/common'

@Injectable()
export class AppService { 
  start(): void {
    Logger.log(`${AppService.name} has started.`, AppService.name)    // this will be displayed
  }
}
```

## Expected behavior
The expected behaviour is that the log level is applied to the static Logger as well.

## Possible Solution
Include a log level check in the static Logger log functions.

## Environment

<pre><code>
Nest version: 7.4.2 and the most recent version also.
 </code></pre>

**Comment by jmcdo29** [comment]
Can you provide a minimum reproduction?

**Comment by phils-hub** [comment]
Hi @jmcdo29 is it really necessary? I took a look at _packages/common/services/logger.service.ts_ and its quite clear that there is no equivalent function for `isLogLevelEnabled`. Let me know if you still need one 👍

**Comment by jmcdo29** [comment]
You're right that is not the. Generally the static methods of the logger shouldn't be used though, for that specific reason. Nest has then there for some internal logs, and even then I think most of them have been removed. I'll need to run a grep against the code to be sure. 

Using the static methods mashed mocking more difficult. Do you have a specific reason not to use an instance of the logger?

**Comment by phils-hub** [comment]
Well, i used the static logger to avoid having to inject an instance, as the logger is "just" a utility for me. It was a matter of convenience at the time i would say. I can of course switch to injecting a (singleton) instance of the logger. If you do not see this as a bug then we can close this issue. However, it would perhaps be good to highlight somewhere in the class description that the static log methods do not take the log level into account. It took me quite some time to figure out that the log levels were not working because they simply were not being applied.

**Comment by jmcdo29** [comment]
I think a mention in the docs would be good. Would you like to add a PR for that? Personally, I stay away from `static` unless necessary (things like `forRoot/forRootAsync` and similar module registration methods). They tend to make mocking methods more difficult. Also, if you decide to swap out loggers, there's a chance that the new logger won't have the same static methods, but they should have the same instance methods.

**Comment by matt-lethargic** [comment]
What would be good is some documentation around how to use the built in Logger, I was just looking in the docs and all it mentions is customising the Logger, currently I'd just like to use the vanilla NestJS logging until my needs change, could we have this added to the documentation (at the top of the logging section)?
